frontend/ui/elements/common_settings_menu_table.lua

```diff
@@ -541,14 +542,23 @@ common_settings.document = {
+- alongside the book file itself (the long time default): sdr folders will be visible when you browse your library directories with another file browser or from your computer, which may clutter your vision of your library. But this allows you to move them along when you reorganize your library, and also survives any renaming of parent directories. Also, if you perform directory synchronization or backups, your settings will be part of them.
+- all in koreader/docsettings/: sdr folders will only be visible and used by KOReader, and won't clutter your vision of your library directories with another file browser or from your computer. But any reorganisation of your library (directories or filename moves and renamings) may result in KOReader not finding your previous settings for these books. These settings won't be part of any synchronization or backups of your library.
+- all inside koreader/hashdocsettings/ as hashes: sdr folders are identified not by filepath/filename but by partial MD5 hash, allowing you to rename, move, and copy documents outside of KOReader without sdr folder clutter while keeping them linked to their metadata. However, any file modifications such as writing highlights into PDFs or downloading from Calibre may change the hash, and thus lose their linked metadata. Calculating file hashes may also slow down file browser navigation. This option may suit users with multiple copies of documents across different devices and directories.]])
+
+
+local hash_filemod_warn = _([[Warning: /koreader/hashdocsettings requires calculating partial file hashes of documents which may slow down file browser navigation. Any file modifications (such as embedding annotations into PDF files or downloading from Calibre) may change its partial hash, resulting in the metadata being unlinked from the file and thereby losing its highlights, bookmarks, and progress. Embedding PDF annotations is currently set to "%s" and can be disabled at (Cog -> Document -> Save Document (write highlights into PDF)).]])
+local leaving_hash_sdr_warn = T(_("Warning: You currently have documents with hash-based metadata. Until this metadata is moved by opening those documents, or deleted, file browser navigation may remain slower."))
+local hash_metadata_file_list_header = _([[
+Hash-based metadata has been saved in koreader/hashdocsettings/ for the following documents. Hash-based storage may slow down file browser navigation in large directories. Thus, if not using hash-based metadata storage, it is recommended to open the associated documents in KOReader to automatically migrate their metadata to the preferred storage location, or to delete koreader/hashdocsettings/, which will speed up file browser navigation.
```

Comment:
`Hash-based metadata has been saved`, or `have been saved` plural?

Reply:
Data and metadata are both technically plural but are more commonly used as singular in my experience. I swap between them regularly and can do whatever KOReader does - let me know your preference.

Reply:
Not a native English speaker - so I can't have/be a p/reference :)
@Frenzie ?

Reply:
It's used as both singular and plural. I think we probably mainly use (meta)data as singular, but in this case it's arguably also a bit of a quantifying expression on the semantically and grammatically plural documents. I.e., how English speakers say "a couple of people _are_" instead of "a couple of people _is_".

tl;dr singular's fine.

But all these strings here have many words; I'll have to look at that in more detail later. I'd like to imagine a brief précis of their meaning is attainable.

Reply:
Sounds good, resolving this one but I'll await wording feedback from @Frenzie later if you have any.